cocotb/scheduler.py

```diff
@@ -628,51 +630,64 @@ async def wrapper():
 
         return wrapper()
 
-    def add(self, coroutine):
-        """Add a new coroutine.
-
-        Just a wrapper around self.schedule which provides some debug and
-        useful error messages in the event of common gotchas.
-        """
+    @staticmethod
+    def create_task(coroutine: Any) -> None:
+        """ Checks to see if the given object is a schedulable coroutine object """
 
+        if isinstance(coroutine, cocotb.decorators.RunningTask):
+            return coroutine
+        if inspect.iscoroutine(coroutine):
+            return cocotb.decorators.RunningTask(coroutine)
         if inspect.iscoroutinefunction(coroutine):
             raise TypeError(
                 "Coroutine function {} should be called prior to being "
                 "scheduled."
                 .format(coroutine))
-
         if isinstance(coroutine, cocotb.decorators.coroutine):
             raise TypeError(
                 "Attempt to schedule a coroutine that hasn't started: {}.\n"
                 "Did you forget to add parentheses to the @cocotb.test() "
                 "decorator?"
                 .format(coroutine)
             )
-
-        if inspect.iscoroutine(coroutine):
-            return self.add(cocotb.decorators.RunningTask(coroutine))
-
-        elif sys.version_info >= (3, 6) and inspect.isasyncgen(coroutine):
+        if sys.version_info >= (3, 6) and inspect.isasyncgen(coroutine):
             raise TypeError(
                 "{} is an async generator, not a coroutine. "
                 "You likely used the yield keyword instead of await.".format(
                     coroutine.__qualname__))
+        raise TypeError(
+            "Attempt to add a object of type {} to the scheduler, which "
+            "isn't a coroutine: {!r}\n"
+            "Did you forget to use the @cocotb.coroutine decorator?"
+            .format(type(coroutine), coroutine)
+        )
 
-        elif not isinstance(coroutine, cocotb.decorators.RunningTask):
-            raise TypeError(
-                "Attempt to add a object of type {} to the scheduler, which "
-                "isn't a coroutine: {!r}\n"
-                "Did you forget to use the @cocotb.coroutine decorator?"
-                .format(type(coroutine), coroutine)
-            )
+    def add(self, coroutine):
```

Comment:
Might as well add annotations.

Reply:
What do you think about adding the name alias `run_now` for `add`?

Reply:
I think we need to figure out what we want the **new** interface to be, and hopefully make it coherent. We don't have to necessarily use `fork` going forward, we can keep it for backwards compatibility and maybe deprecate it eventually. Relates to #1526.

As this PR is now, we have:
- `cocotb.fork` -> `scheduler.add`
- `cocotb.fork_soon` -> `scheduler.run_later`

Reply:
Alright, I'll actually remove the `fork_soon` proxy, and leave everything else as it is.